src/ghazal_azadi_ex/ex03_project/test_sorting.py

```diff
@@ -0,0 +1,98 @@
+__author__ = 'Ghazal_Azadi'
+__email__ = '@ghazal.azadi@nmbu.no'
+
+
+def bubble_sort(my_data):
+    # The main function which tests would be performed on it afterwards
+    my_data = list(my_data)
+    # Changing type of data to list which becomes mutable
+    dim = len(my_data)
+    made_swap = True
+    swaps = 0
+    while made_swap:
+        made_swap = False
+        for cnt in range(dim-1):
+            if my_data[cnt] > my_data[cnt+1]:
+                my_data[cnt], my_data[cnt+1] = my_data[cnt+1], my_data[cnt]
+            # Comparing elements and swap them if needed.
+                made_swap = True
+                swaps = swaps + 1
+    return my_data
+
+
+def test_empty():
+    """Test that the sorting function works for empty list"""
+    empty_list = []
+    assert bubble_sort(empty_list) == empty_list
+
+
+def test_single():
+    """Test that the sorting function works for single-element list"""
+    assert bubble_sort([2]) == [2]
+
+
+def test_sorted_is_not_original():
+    """
+    Test that the sorting function returns a new object.
+
+    Consider
+
+    data = [3, 2, 1]
+    sorted_data = bubble_sort(data)
+
+    Now sorted_data shall be a different object than data,
+    not just another name for the same object.
+    """
+    data = [3, 2, 1]
+    assert bubble_sort(data) != data
+
+
+def test_original_unchanged():
+    """
+    Test that sorting leaves the original data unchanged.
+
+    Consider
+
+    data = [3, 2, 1]
+    sorted_data = bubble_sort(data)
+
+    Now data shall still contain [3, 2, 1].
+    """
+    data = [3, 2, 1]
+    sorted_data = bubble_sort(data)
+    assert data == [3, 2, 1]
+
+
+def test_sort_sorted():
+    """Test that sorting works on sorted data."""
+    data = [3, 2, 1]
+    sorted_data = bubble_sort(data)
```

Comment:
You are testing bubble sort, there's no proof that bubble_sort can sort the data correctly so that you can use here.
1. Remove this line and hard-code data into a sorted-list OR
2. Use the built-in python function for sorting: `sorted(data)`